lib/sidekiq/encrypted_args.rb

```diff
@@ -79,25 +82,29 @@ def decrypt(encrypted_data)
       # Helper method to get the encrypted args option from an options hash. The value of this option
       # can be `true` or an array indicating if each positional argument should be encrypted, or a hash
       # with keys for the argument position and true as the value.
-      def encrypted_args_option(worker_class)
-        sidekiq_options = worker_class.sidekiq_options
-        option = sidekiq_options.fetch(:encrypted_args, sidekiq_options["encrypted_args"])
-
+      def encrypted_args_option(sidekiq_options, worker_class, args)
+        option = sidekiq_options.fetch("encrypted_args", sidekiq_options[:encrypted_args])
         return nil if option.nil?
+        return [] if option == false
 
-        return Hash.new(true) if option == true
-
-        return replace_argument_positions(worker_class, option) if option.is_a?(Hash)
-
-        hash = {}
-        Array(option).each_with_index do |val, position|
-          if val.is_a?(Symbol) || val.is_a?(String)
-            hash[val] = true
-          else
-            hash[position] = val
+        indexes = []
+        if option == true
+          args.size.times { |i| indexes << i }
+        elsif option.is_a?(Hash)
+          indexes = replace_argument_positions(worker_class, option)
+        else
+          Array(option).each_with_index do |val, position|
+            if val.is_a?(Integer)
+              indexes << val
+            elsif val.is_a?(Symbol) || val.is_a?(String)
+              position = perform_method_parameter_index(worker_class, val)
+              indexes << position if position
+            elsif val
+              indexes << position
+            end
           end
```

Comment:
We might want to enforce that the array must be all values of one type of specification (Array<Bool>, Array<String, Symbol>, Array<Integer>). Currently, it is valid to do `["arg1", false, 1, true]`.

Also, if they pass in an array of booleans, we should enforce that it matches the size of the args array.